### PR TITLE
feat: watchdog delay

### PR DESCRIPTION
**Pull request by madvosburg**
Added a 3 second delay on start of program for watchdog to begin as requested by Joel.

Also playing around with original sampling idea in main loop for button presses.

### CODE REVIEW

**Comment by JMenterZXT** [comment]
@madvosburg how is the operation with the new board? Is there also any difference/benefit with this new branch?

File: Core/Src/main.c [hunk 1]
```diff
@@ -69,41 +69,48 @@ static void MX_WWDG_Init(void);
...
+	BUTTON_4,
+	//could track up to 32 buttons
+	NONE_B
 }button_state;
 
 button_state button = NONE_B;
 GPIO_PinState state;
+uint32_t button_flag = 0;
+
+uint32_t previousMillis = 0;
```
Comment (alextran1502) [inline Core/Src/main.c:88]: You should keep it consistent with variable casing, so it should 
```suggestion
uint32_t previous_millis = 0;
```

File: Core/Src/main.c [hunk 2]
```diff
@@ -175,385 +182,360 @@ void sendData(){
...
-		if(HAL_GPIO_ReadPin(GPIOC, GPIO_PIN_3) == state && button == BUTTON_2){
-			bitmask_set(1);
-		}
-		if(HAL_GPIO_ReadPin(GPIOB, GPIO_PIN_14) == state && button == BUTTON_3){
-			bitmask_set(2);
-		}
-		if(HAL_GPIO_ReadPin(GPIOB, GPIO_PIN_15) == state && button == BUTTON_4){
-			bitmask_set(3);
-		}
+		timer_flag = true;
```
Comment (alextran1502) [inline Core/Src/main.c:504]: How come this code change is related to watchdog functionality?

**Comment by madvosburg** [comment]
@JMenterZXT Looks like the new board is working well. Did some testing with just the multimeter to make sure the hardware was not the issue. I found that this new code had less flickering issues- the visual bouncing could only be seen once in a while.

File: Core/Src/main.c (hunk 1 above)
Comment (madvosburg) [reply Core/Src/main.c:88]: Great point, I will change those now

File: Core/Src/main.c (hunk 2 above)
Comment (madvosburg) [reply Core/Src/main.c:504]: I deleted the use of the original timer for button debouncing so I reused that timer16 for the watchdog delay. Specifically, I use that flag to send the watchdog refresh after the timer is called back (ie 3 second delay into the program)

**Pull request merged**